pkg/pointerdb/pdbclient/client.go

```diff
@@ -48,21 +48,30 @@ type Client interface {
 	Delete(ctx context.Context, path p.Path) error
 }
 
+func apiKeyInjector(APIKey string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		ctx = metadata.AppendToOutgoingContext(ctx, "apikey", APIKey)
+
+		var header metadata.MD
+		opts = append(opts, grpc.Header(&header))
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		// TODO do something with header
```

Comment:
Header is passed to invoker and it will be (or not) filled with a response from server. In this case after invoker execution header should be filled with some data (e.g. signature)